core/capabilities/ccip/ccipsolana/commitcodec.go

```diff
@@ -0,0 +1,141 @@
+package ccipsolana
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+
+	agbinary "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// CommitPluginCodecV1 is a codec for encoding and decoding commit plugin reports.
+// Compatible with:
+// - "OffRamp 1.6.0-dev"
+type CommitPluginCodecV1 struct{}
+
+func NewCommitPluginCodecV1() *CommitPluginCodecV1 {
+	return &CommitPluginCodecV1{}
+}
+
+func (c *CommitPluginCodecV1) Encode(ctx context.Context, report cciptypes.CommitPluginReport) ([]byte, error) {
+	var buf bytes.Buffer
+	encoder := agbinary.NewBorshEncoder(&buf)
+	if len(report.MerkleRoots) == 0 || len(report.MerkleRoots) > 1 {
```

Comment:
```suggestion
	if len(report.MerkleRoots) != 1 {
```

We want exactly one merkle root right? 

Side note: why are we enforcing only one merkle root in the commit report? Is there some Solana-specific limitation for commit reports as well, meaning we can't do multi-source batching of roots? cc @winder @archseer